### PR TITLE
Reduce crt heap diff

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

Reduce differences of heap.c to wine (version 5.0rc5)
This basically reverts most of the nasty hacks that have been applied.
The goal is to make the file auto-syncable with the winesync script.
It also serves as a lesson for everyone how to make 3rd-party code compile and work without hacking every other line.

## Proposed changes
- Get back MSVCRT prefix
- Use the original wine heap functions
- Remove some unneccessary changes
- Add some functions from wine
- Sync some wine commits

## Tests
https://reactos.org/testman/compare.php?ids=85614,85621,85626,85628,85630

### CODE REVIEW

File: sdk/lib/crt/wine/heap.c [hunk 1]
```diff
@@ -703,11 +703,9 @@ int CDECL MSVCRT_strncpy_s(char *dest, MSVCRT_size_t numberOfElements,
     if(!count)
         return 0;
 
-    if (!MSVCRT_CHECK_PMT(dest != NULL) || !MSVCRT_CHECK_PMT(src != NULL) ||
-        !MSVCRT_CHECK_PMT(numberOfElements != 0)) {
-        *_errno() = EINVAL;
```
Comment (HBelusca) [inline sdk/lib/crt/wine/heap.c:589]: Shouldn't errno be updated as well?

**Review by HBelusca — APPROVED** [review]
Dunno why both msi:msi and user32:menu wine tests appear to generate one or two more errors, but otherwise this looks fine.

Sad that Wine had to pollute their code with those "MSVCRT_" prefixes...

File: sdk/lib/crt/wine/heap.c [hunk 2]
```diff
@@ -35,51 +60,131 @@
...
+static HANDLE heap, sb_heap;
 
-typedef int (CDECL *MSVCRT_new_handler_func)(size_t size);
+typedef int (CDECL *MSVCRT_new_handler_func)(MSVCRT_size_t size);
 
 static MSVCRT_new_handler_func MSVCRT_new_handler;
 static int MSVCRT_new_mode;
 
-/* FIXME - According to documentation it should be 8*1024, at runtime it returns 16 */
+/* FIXME - According to documentation it should be 8*1024, at runtime it returns 16 */
```
Comment (binarymaster) [inline sdk/lib/crt/wine/heap.c:72]: image

Comment (HBelusca) [reply sdk/lib/crt/wine/heap.c:72]: Potentially said 3rd-party (wine) code had a trailing whitespace?

Comment (binarymaster) [reply sdk/lib/crt/wine/heap.c:72]: It's okay in this case, which I didn't checked. :sweat_smile:

**Comment by tkreuzer** [comment]
> Dunno why both msi:msi and user32:menu wine tests appear to generate one or two more errors, but otherwise this looks fine.

Those are flaky.

> Sad that Wine had to pollute their code with those "MSVCRT_" prefixes...

In fact they removed them. They will go away after syncing the file.

File: sdk/lib/crt/wine/heap.c (hunk 1 above)
Comment (tkreuzer) [reply sdk/lib/crt/wine/heap.c:589]: It's updated in MSVCRT_CHECK_PMT

**Comment by JoachimHenze** [comment]
Attention, one of the commits in this PR did introduce regression https://jira.reactos.org/browse/CORE-19505

**Pull request merged**